geopandas/io/tests/test_file.py

```diff
@@ -1298,6 +1297,51 @@ def test_write_read_file(test_file, engine):
     os.remove(os.path.expanduser(test_file))
 
 
+@pytest.mark.skipif(fiona is False, reason="Fiona not available")
+@pytest.mark.skipif(FIONA_GE_19, reason="Fiona >= 1.9 supports metadata")
+def test_to_file_metadata_unsupported_fiona_version(tmp_path, df_points):
+    metadata = {"title": "test"}
+    tmp_file = tmp_path / "test.gpkg"
+    match = "'metadata' keyword is only supported for Fiona >= 1.9"
+    with pytest.raises(NotImplementedError, match=match):
+        df_points.to_file(tmp_file, driver="GPKG", engine="fiona", metadata=metadata)
+
+
+@pytest.mark.skipif(not FIONA_GE_19, reason="only Fiona >= 1.9 supports metadata")
+def test_to_file_metadata_supported_fiona_version(tmp_path, df_points):
+    metadata = {"title": "test"}
+    tmp_file = tmp_path / "test.gpkg"
+
+    df_points.to_file(tmp_file, driver="GPKG", engine="fiona", metadata=metadata)
+
+    # Check that metadata is written to the file
+    with fiona.open(tmp_file) as src:
+        tags = src.tags()
+        assert tags == metadata
+
+
+@pytest.mark.skipif(pyogrio is False, reason="Pyogrio not available")
+def test_to_file_metadata_unsupported_pyogrio_version(tmp_path, df_points):
```

Comment:
This doesn't have a pyogrio version guard, it seems like it should be failing now for most of the envs that have pyogrio.

Please also consider adding a test for a supported version of pyogrio.

Reply:
Ah whoops, I've deleted the wrong test! The last commit I took out the version guards since all supported pyogrio version support metadata

Reply:
@brendan-ward sorry about that, should be fixed now, but you're right it would have shown up if the tests had run (I didn't look at them on CI, just assumed they were fine since I didn't get a fail email). I don't know why but it looks like only the docs have been running on the last few commits here